src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -81,6 +81,12 @@ public function createView(ChoiceListInterface $list, $preferredChoices = null,
                     $otherViews
                 );
             }
+
+            if ($groupByOrder && is_array($groupByOrder)) {
+                uksort($otherViews, function ($key1, $key2) use ($groupByOrder) {
```

Comment:
this logic does not seem to support callable values for `$groupByOrder`, while the interface allows it (and callables can be arrays, making things even worse).